src/simulators/superoperator/superoperator.hpp

```diff
@@ -0,0 +1,152 @@
+/**
+ * This code is part of Qiskit.
+ *
+ * (C) Copyright IBM 2018, 2019.
+ *
+ * This code is licensed under the Apache License, Version 2.0. You may
+ * obtain a copy of this license in the LICENSE.txt file in the root directory
+ * of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Any modifications or derivative works of this code must retain this
+ * copyright notice, and modified files need to carry a notice indicating
+ * that they have been altered from the originals.
+ */
+
+#ifndef _qv_superoperator_hpp_
+#define _qv_superoperator_hpp_
+
+
+#include "framework/utils.hpp"
+#include "simulators/densitymatrix/densitymatrix.hpp"
+
+namespace QV {
+
+//============================================================================
+// Superoperator class
+//============================================================================
+
+// This class is derived from the DensityMatrix class and stores an N-qubit 
+// superoperator as a 2 * N-qubit vector.
+// The vector is formed using column-stacking vectorization of the
+// superoperator (itself with respect to column-stacking vectorization).
+
+template <typename data_t = double>
+class Superoperator : public DensityMatrix<data_t> {
+
+public:
+  // Parent class aliases
+  using BaseVector = QubitVector<data_t>;
+  using BaseDensity = DensityMatrix<data_t>;
+  using BaseUnitary = UnitaryMatrix<data_t>;
+
+  //-----------------------------------------------------------------------
+  // Constructors and Destructor
+  //-----------------------------------------------------------------------
+
+  Superoperator() : Superoperator(0) {};
+  explicit Superoperator(size_t num_qubits);
```

Comment:
What implicit conversion are we trying to avoid here? ... not sure if explicit makes sense.. but in any case, is not bothering.

Reply:
we can assing `num_qubits_` in the initializer list of the ctor: 
```c++
explicit Superoperator(size_t num_qubits) : num_qubits_(set_num_qubits(num_qubits)){
 }
```
this used to save some time at object construction time (at least the last time I saw it, It may not be the case now)

Reply:
This doesn't compile for me